### DISCUSSION TITLE
Minion: is there a way to programmatically stop the loop?

### DISCUSSION THREAD

**Discussion by fluca1978** [mojolicious/mojo#2189] [Q&A]
I've a Perl application that is using `Minion` to enqueue a set of tasks, then calling `$minion->worker->run`.
But I would like to stop the main loop once all the tasks have completed (successfully). In other words, I would like `run` to work pretty much as `$minion->perform_tasks` but with parallelism. Is that possible?

**Comment by kraih** [comment]
See how `perform_jobs` is implemented. https://docs.mojolicious.org/Minion#perform_jobs

**Reply by fluca1978** [reply]
> See how `perform_jobs` is implemented. https://docs.mojolicious.org/Minion#perform_jobs

Thanks, so the trick is to `unregister` the worker after having dequeued the jobs. However, if I'm right, to start in parallel, I need something like `Minion::Job::start` instead of `Minion::Job::Perform`, so:

```
my $worker = $minion->worker;
while (my $job = $worker->register->dequeue(0)) { $job->start }
$worker->unregister;
```

(assuming I don't have so much tasks to fill up the CPU)
